### PR TITLE
Configure PWM ClockFrequency (fix towards V2.7.7 release)

### PR DESCRIPTION
**Pull request by brentpicasso**
turns out that the V2 firmware had no way of changing the PWM clock frequency. reported by customer. Default PWM clock frequency is 20KHz.

Fixed so PWM clock frequency can be updated via Lua Script. Later, clock frequency setting will be configurable via PWM configuration screen in app. 

Some code cleanup in areas that were touched.

Bench tested on MK1 and MK2

### CODE REVIEW

File: include/logger/loggerConfig.h [hunk 1]
```diff
@@ -260,9 +260,9 @@ typedef struct _PWMConfig{
 } PWMConfig;
 
 /// PWM frequency in Hz.
-#define MAX_PWM_CLOCK_FREQUENCY             200000
-#define MIN_PWM_CLOCK_FREQUENCY	 	10
-#define DEFAULT_PWM_CLOCK_FREQUENCY			10000
+#define MAX_PWM_CLOCK_FREQUENCY             65000
```
Comment (stieg) [inline include/logger/loggerConfig.h]: https://www.kernel.org/doc/Documentation/CodingStyle : Chapter 12

Please make it one space between the end of the definition and the value

```#define MAX_PWM_CLOCK_FREQUENCY 65000```

File: include/logger/loggerConfig.h [hunk 2]
```diff
@@ -527,7 +527,7 @@ char filterPwmLoggingMode(int config);
 unsigned short filterPwmDutyCycle(int dutyCycle);
 unsigned short filterPwmPeriod(int period);
 int filterImuRawValue(int accelRawValue);
-int filterPwmClockFrequency(int frequency);
+uint16_t filterPwmClockFrequency(uint16_t frequency);
```
Comment (stieg) [inline include/logger/loggerConfig.h]: :+1:

File: src/logger/luaLoggerBinding.c [hunk 3]
```diff
@@ -128,7 +128,11 @@ int Lua_GetAtSplit(lua_State *L){
 
 int Lua_SetPWMClockFrequency(lua_State *L){
 	if (lua_gettop(L) >= 1){
-		getWorkingLoggerConfig()->PWMClockFrequency = filterPwmClockFrequency(lua_tointeger(L,1));
+		LoggerConfig *loggerConfig = getWorkingLoggerConfig();
+
+		uint16_t clock_frequency = filterPwmClockFrequency(lua_tointeger(L,1));
+		loggerConfig->PWMClockFrequency = clock_frequency;
+		PWM_set_clock_frequency(clock_frequency);
```
Comment (stieg) [inline src/logger/luaLoggerBinding.c]: Should the `PWM_set_clock_frequency` method also handle updating the LoggerConfig too?  Seems like it would be better bundled that way unless there are cases where you would want to change one but not the other.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c [hunk 4]
```diff
@@ -10,7 +10,11 @@
 #define MAX_DUTY_CYCLE 100
 #define PWM_CHANNEL_COUNT 4
 
-#define DEFAULT_PERIOD 1000
+#define CLOCK_FREQUENCY_PRESCALER_SCALING 42000000
+
+//magic number 20970 = trimmmed value.
+//adjust to calibrate further (if using a more accurate scope)
+#define CLOCK_FREQUENCY_PERIOD_SCALING 20970
```
Comment (stieg) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: :100: That is the correct way per our standard.

File: include/logger/loggerConfig.h (hunk 1 above)
Comment (Jeff-Ciesielski) [inline include/logger/loggerConfig.h]: I noticed a lot of non-kernel style violations as well, but ignored them for the moment.  Is better to go through and address style issues at once rather than ad-hoc to avoid inconsistency IMO.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c [hunk 5]
```diff
@@ -37,6 +41,8 @@ static gpio analogCtrlGpios[] = {
 	{ RCC_AHB1Periph_GPIOE, GPIOE, GPIO_Pin_1}
 };
 
+static uint16_t g_currentClockPeriod = 0;
```
Comment (stieg) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: the `= 0` is not needed here as that is the default value of static variables as defined per BSS rules on ELF files.  The = 0 makes the firmware larger for no good reason (albeit a very small size increase).

File: src/logger/luaLoggerBinding.c (hunk 3 above)
Comment (stieg) [inline src/logger/luaLoggerBinding.c]: Hrmm... I'm on the fence about this one.  Your call here.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c [hunk 6]
```diff
@@ -69,14 +75,16 @@ static void initAnalogControlGpios(){
...
 
-	uint16_t prescaler = (uint16_t) ((SystemCoreClock) / 210000) - 1;
-	prescaler = 4 - 1;
+	return 1;
+}
+
+void PWM_device_set_clock_frequency(unsigned short clockFrequency){
+
+	uint32_t period = (1000 * ((CLOCK_FREQUENCY_PERIOD_SCALING * 10000)/clockFrequency)) / 10000;
+	uint16_t prescaler = (uint16_t) ((SystemCoreClock) / CLOCK_FREQUENCY_PRESCALER_SCALING) - 1;
```
Comment (stieg) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: Use `const` for code safety.

File: src/logger/luaLoggerBinding.c (hunk 3 above)
Comment (brentpicasso) [inline src/logger/luaLoggerBinding.c]: No, it shouldn't based on the pattern and style of the rest of the HAL layer.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c (hunk 4 above)
Comment (brentpicasso) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: so is that a thumbs up?

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c (hunk 6 above)
Comment (brentpicasso) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: for which lines / variables?

File: include/logger/loggerConfig.h (hunk 1 above)
Comment (brentpicasso) [inline include/logger/loggerConfig.h]: this is no different style than what it was before. Although, I am touching other things elsewhere as I go (stdint 's  for example).

Comment (stieg) [inline include/logger/loggerConfig.h]: My rule has always been that if you touch it, you are responsible for make it fit to style.  It promotes moving towards the standard.  I like @Jeff-Ciesielski 's approach, but until that is done I think my policy is the way to move forward.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c (hunk 4 above)
Comment (stieg) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: yes.

File: stm32_base/hal/PWM_stm32/PWM_device_stm32.c (hunk 6 above)
Comment (stieg) [inline stm32_base/hal/PWM_stm32/PWM_device_stm32.c]: Period and prescaler.

File: include/logger/loggerConfig.h (hunk 1 above)
Comment (Jeff-Ciesielski) [inline include/logger/loggerConfig.h]: I disagree, if we do that, then we have multiple non-related commits polluting the pull request.  It's a lot of noise.

Comment (brentpicasso) [inline include/logger/loggerConfig.h]: I don't see where in chapter 12 of that document it calls for one space between end of definition and value. What am I missing?

Comment (stieg) [inline include/logger/loggerConfig.h]: I don't see it mentioned explicitly either... but I remember that being the standard.

@Jeff-Ciesielski,
   I don't follow your logic.  Perhaps let me rephrase my point:

If the person changing the code is _touching_ the lines anyways (as @brentpicasso was doing in this case), then any lines he touches should be brought up to code standard.  I'm not implying doing that for the whole block per the reason you just mentioned.  Just the lines that are touched.

Comment (brentpicasso) [inline include/logger/loggerConfig.h]: A couple of things:
 
1) I don't see in the kernel coding guide a recommendation on this topic
2) if there is little risk to bugs, I would like to make this visual formatting an exception to the style guide. I really like seeing the visual formatting.
